### DISCUSSION TITLE
[prettier-plugin-blade] How to prevent html tag on a new line

### DISCUSSION THREAD

**Discussion by kjvdven** [Stillat/blade-parser-typescript#59] [Q&A]
When I save a blade.php file with this plugin, my html opening tags are all put on a new line. 
Ideally I would only like that only automatic for block elements, not for the inline elements. 

Are there options in prettier-plugin-blade I can set so that prettier doesn't place all elements on a new line?

When I have:

```html
<p>
  Lorem ipsum dolor sit, amet consectetur adipisicing elit. <strong>Similique accusantium</strong> veniam
  accusamus ducimus, eveniet aliquid <u>atque reiciendis</u> beatae illum amet fugiat odit fuga <em>nobis
  commodi</em> cupiditate facere distinctio modi sed.
</p>
```

After save it becomes:

```html
<p>
  Lorem ipsum dolor sit, amet consectetur adipisicing elit.
  <strong>Similique accusantium</strong>
  veniam accusamus ducimus, eveniet aliquid
  <u>atque reiciendis</u>
  beatae illum amet fugiat odit fuga
  <em>nobis commodi</em>
  cupiditate facere distinctio modi sed.
</p>
```

I just want to keep what I have.

**Comment by JohnathonKoster** [comment]
This seems related to how it manages the `htmlWhitespaceSensitivity` internally. The plugin needs to manage this setting itself at the moment in order to produce the correct Blade structure.

Will need to think on this - **Update**: I can support the other whitespace mode that will help achieve what you want, it will just take me a while to get things updated and tested (unfortunately it won't be anytime within the next few weeks or so as I need to work out some layout challenges in a wide number of situations 🙂)

**Reply by kjvdven** [reply]
That's awesome! 

If you need a hand testing, I'm glad to help.

**Comment by stayallive** [comment]
I would also like this to be possible since it currently also introduces extra spaces which shouldn't be part of the content. 

Considering:

```html
Shorten (and unshorten) URLs with your favorite automation tools like 
<a href="#" target="_blank" rel="noopener">Make</a> (formerly Integromat),
<a href="#" target="_blank" rel="noopener">Raycast</a>,
<a href="#" target="_blank" rel="noopener">Zapier</a>,
```

Turns into:

```html
Shorten (and unshorten) URLs with your favorite automation tools like
<a href="#" target="_blank" rel="noopener">Make</a>
(formerly Integromat),
<a href="#" target="_blank" rel="noopener">Raycast</a>
,
<a href="#" target="_blank" rel="noopener">Zapier</a>
,
```

This introduces extra spaces in the output so it changes my content which is not ideal and unexpected.

As a workaround I wrapped my content in `{{-- format-ignore-start --}}...{{-- format-ignore-end --}}` for now (which I'm very glad exists ❤️) but it would be nice especially on larger projects to use this without worrying it might change my actual output instead of just the code formatting 😄

**Reply by JohnathonKoster** [reply]
Definitely understand that scenario. It's something I've been actively looking into, but still quite a few challenges to overcome before I feel comfortable making that configurable at this time

**Reply by JohnathonKoster** [reply]
Just a small progress update: I've got a version in the works that is handling changes to that setting well. I still need to do a ton of additional testing, but won't be *too* much longer 🙂

**Reply by stayallive** [reply]
This is very encouraging, thanks for the hard work you are putting into this!

**Reply by MartinPezet** [reply]
@JohnathonKoster Did this change mentioned here ever get released? If not then what branch was this change created on?

**Reply by JohnathonKoster** [reply]
@MartinPezet  No, not yet. Quite a few changes involved quite a few changes to prettier itself. Currently assessing options in the background with others (hopefully more to come soon)

**Reply by MartinPezet** [reply]
Great to hear. Thanks so much for creating this plugin! It's really great!
